### DISCUSSION TITLE
Help Simplifying

### DISCUSSION THREAD

**Discussion by mikekuntz43** [ImageMagick/ImageMagick#4886] [Help] [answered]
I am very new to this. The code below does what I want it to do, but I feel like it is excessively writing the file to disk, Is there any way to simplify this?  Thanks! 

set File=%1

magick %File% -resize 294x294 %File%
magick -size 294x294 xc:none -fill %File% -draw "circle 147,147 147,1" %File%
magick %File% -background none -gravity center -extent 330x330 %File%
magick %File% stroke/stroke.png -resize 330x330 -compose over -composite %File%

**Comment by snibgo** [comment, marked as answer]
What is your platform? I assume Windows BAT. You could do something like this. This is untested. Make one change at a time and ensure it still has the expected output.
```
magick ^
  %File% ^
  -resize 294x294 ^
  +write mpr:RES ^
  +delete ^
  -size 294x294 xc:none -fill mpr:RES -draw "circle 147,147 147,1" ^
  -background none -gravity center -extent 330x330 ^
  stroke/stroke.png -resize 330x330 -compose over -composite ^
  %File%
```

**Reply by mikekuntz43** [reply]
That worked perfectly! This saves me so much time as I am applying this to about 100000 files. Thank you!!!
